### PR TITLE
Add temperature dependant load profile

### LINKED ISSUE #601: Implementation of temperature dependent load profiles
For that we als need to change load input attribute `standardLoadProfile` to `loadProfile` to address both types of profiles

### PR DESCRIPTION
**Pull request by t-ober**
Resolves #601

### CODE REVIEW

File: src/main/java/edu/ie3/datamodel/models/TemperatureDependantLoadProfile.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+package edu.ie3.datamodel.models;
+
+/**
+ * Temperature dependant load profiles for night storage heating and heat pumps . The profiles rely
+ * on the VDN description for interruptable loads. For more details see <a
+ * href="https://www.bdew.de/media/documents/LPuVe-Praxisleitfaden.pdf">here</a>.
+ *
+ * <p>These are the ones determined by NBW (accessed 05/2022)
+ */
+public enum TemperatureDependantLoadProfile implements StandardLoadProfile {
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/models/TemperatureDependantLoadProfile.java:15]: Compared to `BdewLoadProfile`, the parsing method `get(String key)` and `toString()` are missing. If you think that you won't need them this is totally find, just a heads up here.

**Review by sebastian-peter — APPROVED** [review]
LGTM

File: src/main/java/edu/ie3/datamodel/models/StandardLoadProfile.java [hunk 2]
```diff
@@ -34,7 +35,9 @@ static StandardLoadProfile parse(String key) throws ParsingException {
     if (key == null || key.isEmpty()) return DefaultLoadProfiles.NO_STANDARD_LOAD_PROFILE;
 
     String filterKey = key.toLowerCase().replaceAll("[-_]*", "");
-    return Arrays.stream(BdewLoadProfile.values())
+    return Stream.concat(
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/models/StandardLoadProfile.java:38]: The exception String in line 43 is not correct anymore

**Review by sebastian-peter — CHANGES_REQUESTED** [review]

**Review by danielfeismann — APPROVED** [review]
LGTM, thanks :)

File: src/main/java/edu/ie3/datamodel/models/profile/LoadProfile.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+   * Parses the given key to {@link StandardLoadProfile}.
+   *
+   * @param key Key to parse
+   * @return Matching {@link StandardLoadProfile}
+   * @throws ParsingException If key cannot be parsed
+   */
+  static LoadProfile parse(String key) throws ParsingException {
+    if (key == null || key.isEmpty()) return LoadProfile.DefaultLoadProfiles.NO_LOAD_PROFILE;
+
+    String filterKey = key.toLowerCase().replaceAll("[-_]*", "");
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/models/profile/LoadProfile.java:28]: This should probably also be used in `getProfile` below for consistency. Maybe find a common method?

File: src/test/groovy/edu/ie3/datamodel/io/naming/FileNamingStrategyTest.groovy [hunk 4]
```diff
@@ -439,7 +439,7 @@ class FileNamingStrategyTest extends Specification {
 
 		where:
 		clazz            | uuid                                                    | type               || expectedFileName
-		LoadProfileInput | UUID.fromString("bee0a8b6-4788-4f18-bf72-be52035f7304") | BdewLoadProfile.G3 || "test_grid" + File.separator + "input" + File.separator + "global" + File.separator + "lpts_g3_bee0a8b6-4788-4f18-bf72-be52035f7304"
+		LoadProfileInput | UUID.fromString("bee0a8b6-4788-4f18-bf72-be52035f7304") | BdewStandardLoadProfile.G3 || "test_grid" + File.separator + "input" + File.separator + "global" + File.separator + "lpts_g3_bee0a8b6-4788-4f18-bf72-be52035f7304"
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/io/naming/FileNamingStrategyTest.groovy:442]: Indentation

File: src/test/groovy/edu/ie3/datamodel/models/LoadProfileTest.groovy [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+	def "Nbw temperature dependant load profiles can be gotten by their key"() {
+		when:
+		NbwTemperatureDependantLoadProfile actual = NbwTemperatureDependantLoadProfile.get(key)
+
+		then:
+		actual == expected
+
+		where:
+		key     || expected
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/models/LoadProfileTest.groovy:117]: Indentation

File: src/test/groovy/edu/ie3/datamodel/models/LoadProfileTest.groovy [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+ */
+package edu.ie3.datamodel.models
+
+import edu.ie3.datamodel.models.profile.BdewStandardLoadProfile
+import edu.ie3.datamodel.models.profile.LoadProfile
+import edu.ie3.datamodel.models.profile.NbwTemperatureDependantLoadProfile
+import spock.lang.Specification
+
+class LoadProfileTest extends Specification {
+	def "Load profiles are parsed correctly from correct input" () {
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/models/LoadProfileTest.groovy:14]: I also would've loved to see a test for throwing the exception when providing an unknown key, but this was not implemented before, so I wouldn't expect you to implement it.  Just if you feel like it...

File: src/test/groovy/edu/ie3/datamodel/utils/validation/SystemParticipantValidationUtilsTest.groovy [hunk 7]
```diff
@@ -326,7 +326,7 @@ class SystemParticipantValidationUtilsTest extends Specification {
 
 		where:
 		invalidLoad                                                                                                                                                    || expectedException
-		SystemParticipantTestData.loadInput.copy().standardLoadProfile(null).build()                                                                                   || new InvalidEntityException("No standard load profile defined for load", invalidLoad)
+		SystemParticipantTestData.loadInput.copy().loadprofile(null).build() || new InvalidEntityException("No standard load profile defined for load", invalidLoad)
```
Comment (sebastian-peter) [inline src/test/groovy/edu/ie3/datamodel/utils/validation/SystemParticipantValidationUtilsTest.groovy:329]: Indentation

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Looks great overall, thanks. I just found some (mostly minor) points that might be worth addressing.

File: CHANGELOG.md [hunk 8]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - SQL time series have a different structure than CSV counterparts [#545](https://github.com/ie3-institute/PowerSystemDataModel/issues/545)
 - Graph with impedance weighted edges including facilities to create it [#440](https://github.com/ie3-institute/PowerSystemDataModel/issues/440)
 - `TimeSeriesTypeSource` providing a source for the mapping of time series uuids to column schemes (previously provided by `TimeSeriesMappingSource`) [#515](https://github.com/ie3-institute/PowerSystemDataModel/issues/515)
+- `TemperatureDependantLoadProfile`s for depiction of profile behavior of night storage heating and heat pumps [#601](https://github.com/ie3-institute/PowerSystemDataModel/issues/601)
```
[the remark this answers is not in the document]
Comment (t-ober) [reply CHANGELOG.md:14]: see line 52 that was my attempt at addressing that

**Review by sebastian-peter — APPROVED** [review]
Thanks!

**Pull request merged**